apps/db/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/db/InntektsmeldingRepository.kt

```diff
@@ -34,43 +35,49 @@ class InntektsmeldingRepository(
             transaction(db) {
                 InntektsmeldingEntitet
                     .select(
-                        InntektsmeldingEntitet.dokument,
                         InntektsmeldingEntitet.skjema,
+                        InntektsmeldingEntitet.dokument,
                         InntektsmeldingEntitet.eksternInntektsmelding,
                         InntektsmeldingEntitet.innsendt,
+                        InntektsmeldingEntitet.avsenderNavn,
                     ).where { InntektsmeldingEntitet.forespoerselId eq forespoerselId.toString() }
                     .orderBy(InntektsmeldingEntitet.innsendt, SortOrder.DESC)
                     .limit(1)
                     .map {
-                        Quadruple(
-                            it[InntektsmeldingEntitet.dokument],
+                        InntektsmeldingResult(
                             it[InntektsmeldingEntitet.skjema],
+                            it[InntektsmeldingEntitet.dokument],
                             it[InntektsmeldingEntitet.eksternInntektsmelding],
                             it[InntektsmeldingEntitet.innsendt],
+                            it[InntektsmeldingEntitet.avsenderNavn],
                         )
                     }
             }.firstOrNull()
                 ?.let { result ->
-                    val inntektsmelding = result.first
-                    val skjema = result.second
-                    val eksternInntektsmelding = result.third
-                    val mottatt = result.fourth
-
                     when {
-                        skjema != null -> LagretInntektsmelding.Skjema(inntektsmelding?.innsenderNavn, skjema.konverterEndringAarsakTilListe(), mottatt)
-                        inntektsmelding != null -> {
+                        result.skjema != null ->
+                            LagretInntektsmelding.Skjema(
+                                avsenderNavn = result.avsenderNavn ?: result.inntektsmeldingGammeltFormat?.innsenderNavn,
```

Comment:
Skal gå over til å bruke `avsenderNavn` istedenfor `inntektsmeldingGammeltFormat?.innsenderNavn` for å frigjøre kvitteringen fra det gamle IM-formatet.